### PR TITLE
Upgrade `ruby` and `bundler` versions

### PR DESCRIPTION
**Pull request by mwolman**
#### Description:
Upgrade `ruby` and `bundler` versions to the latest

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -598,7 +598,8 @@ GEM
     zeitwerk (2.6.16)
 
 PLATFORMS
-  ruby
+  arm64-darwin-23
```
Comment (JulianPasquale) [inline Gemfile.lock:601]: We should probably keep `ruby` as well

File: Gemfile.lock [hunk 2]
```diff
@@ -330,7 +330,6 @@ GEM
     matrix (0.4.2)
     method_source (1.1.0)
     mini_mime (1.1.5)
-    mini_portile2 (2.8.7)
```
Comment (JulianPasquale) [inline Gemfile.lock:333]: This is weird... Do you know why is being removed?

Comment (mwolman) [reply Gemfile.lock:333]: I have no idea 🤷🏻‍♀️ both the `ruby` and this line were removed when I ran `bundle _2.5.14_ install`

File: Gemfile.lock (hunk 1 above)
Comment (mwolman) [reply Gemfile.lock:601]: This line was removed when I ran `bundle _2.5.14_ install` 🤔

Comment (JulianPasquale) [reply Gemfile.lock:601]: Yeah, It's probably due to your system architecture. We have to make sure it works for all of them, so I say let's add it back manually

Comment (mwolman) [reply Gemfile.lock:601]: It's the same problem with `nokogiri`. I see now that it's installing `nokogiri (1.16.6-arm64-darwin)`. But I don't know how to fix it cause if I add the `ruby` line manually when I run `bundle install,` it removes the line again 🥲

Comment (JulianPasquale) [reply Gemfile.lock:601]: Yeah but we still need to fix this. I tried building the Dockerfile locally and got a error due to a missing platform
image

Running `bundle lock --add-platform ruby` adds back the ruby platform and the Docker build starts to work again. I also manually removed the other platforms and only left `ruby` and it works on my local env as well. I think that's probably the best workaround

File: Gemfile.lock (hunk 2 above)
Comment (JulianPasquale) [reply Gemfile.lock:333]: After adding back the platform, this gem was added back to my Gemfile.lock file. I think it's still a dependency but it's not needed on your local platform. We have to make sure we keep it.

**Review by JulianPasquale — APPROVED** [review]

**Review by santib — APPROVED** [review]

**Pull request merged**